src/main/java/com/direwolf20/buildinggadgets/common/schema/template/Template.java

```diff
@@ -197,18 +191,37 @@ public TemplateData next() {
         };
     }
 
+    /**
+     * @return A Template that is normalized to the bounds [(0,0,0), (maxX, maxY, maxZ)]
+     */
+    public Template normalize() {
```

Comment:
Shouldn't it be normalized by default? or is this to re-normalize it and if so when would this be used? Not against the code. Just want to know.

Reply:
A Template may be denormal directly after construction, when it has been deserialized (some Blocks may have gone missing...) (I think I'm able to adoid it when a matrix op is being performed). Both uses are already present.

However I think I should make that Method private, as no one should have to use this outside thid class.